optima/geospatial.py

```diff
@@ -0,0 +1,339 @@
+"""
+GEOSPATIAL
```

Comment:
NB I'm ignoring most of this file as I understand it's been superseded by the next PR.